compaction/compaction.cc

```diff
@@ -1456,8 +1481,14 @@ class scrub_compaction final : public regular_compaction {
         return _scrub_finish_description;
     }
 
-    flat_mutation_reader_v2 make_sstable_reader() const override {
-        auto crawling_reader = _compacting->make_crawling_reader(_schema, _permit, nullptr);
+    flat_mutation_reader_v2 make_sstable_reader(schema_ptr s,
+                                                reader_permit permit,
+                                                const dht::partition_range& range,
+                                                const query::partition_slice& slice,
+                                                tracing::trace_state_ptr trace,
+                                                streamed_mutation::forwarding sm_fwd,
+                                                mutation_reader::forwarding mr_fwd) const override {
+        auto crawling_reader = _compacting->make_crawling_reader(std::move(s), std::move(permit), nullptr);
```

Comment:
Here we ignore the restrictions, should we at least verify that they are for full range?

Reply:
yes. we do something similar in sstable set's single key reader interface.